### PR TITLE
Added Search API functionality

### PR DESCRIPTION
**Pull request by hzargar2**
# Summary

Added Stripe's Search API functionality for charge, customer, invoice, payment_intent, price, product, and subscription.

### Checklist

- [x] ran `cargo make fmt`
- [x] using [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) to hightlight user-facing fixes and features

### CODE REVIEW

**Comment by arlyon** [comment]
Thanks for this, going to make some time to review this properly very soon :)

**Comment by arlyon** [comment]
could you please mark one of the commits with the `feat:` prefix and I will merge

File: src/params.rs [hunk 1]
```diff
@@ -185,6 +185,199 @@ pub struct ListPaginator<T, P> {
     pub params: P,
 }
 
+#[derive(Debug)]
+pub struct SearchListPaginator<T, P> {
```
Comment (arlyon) [inline src/params.rs:189]: I would rather we make ListPaginator be generic over some trait called `PaginableList` which SearchList and List both implement. You can use GATs to have that trait have some generic associated type (the T in this case), that way we don't have duplicated logic for both types of paginator.

**Review by arlyon — CHANGES_REQUESTED** [review]

File: src/params.rs (hunk 1 above)
Comment (hzargar2) [reply src/params.rs]: Hi thanks for the review. I have implemented the suggested changes. I am weak with advanced traits and GAT's so I am not sure if I have implemented it correctly (although I am %99 sure it's fine). Feel free to review it and let me know if I should change anything. Cheers!

**Comment by hzargar2** [comment]
Hi, is there a issue with the CI jobs? Some checks seem to fail but the error is not precise. I don't think the error is from my code. Says `async-stripe` fails to compile but it compiles locally on my machine.

**Comment by arlyon** [comment]
Hi! We have in fact been having issues due to cargo-llvm-cov. I have ripped it out. Please rebase for me and I'll have another look.

**Comment by hzargar2** [comment]
@arlyon I know you might be busy but is there any update on this?  I see it says there is a change requested above which I implemented. Is there anything else you need form my end? Thanks.

**Comment by arlyon** [comment]
Merged, thanks for taking the time, and your patience :)

**Pull request merged**